src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridSmtpTransport.php

```diff
@@ -20,9 +20,9 @@
  */
 class SendgridSmtpTransport extends EsmtpTransport
 {
-    public function __construct(#[\SensitiveParameter] string $key, ?EventDispatcherInterface $dispatcher = null, ?LoggerInterface $logger = null)
+    public function __construct(#[\SensitiveParameter] string $key, ?EventDispatcherInterface $dispatcher = null, ?LoggerInterface $logger = null, private ?string $region = null)
     {
-        parent::__construct('smtp.sendgrid.net', 465, true, $dispatcher, $logger);
+        parent::__construct('null' !== $region ? \sprintf('smtp.%s.sendgrid.net', $region) : 'smtp.sendgrid.net', 465, true, $dispatcher, $logger);
```

Comment:
Sorry to comment on a merged PR, but I'm not sure this works as intended: we noticed mails were not being sent after upgrading to 7.2. Our region was `null` at this point, but the check is performed using `'null'` (as a string). So this ended up as the following error:

> Connection could not be established with host "ssl://smtp..[sendgrid.net:465](http://sendgrid.net:465/)": stream_socket_client(): php_network_getaddresses: getaddrinfo for smtp..[sendgrid.net](http://sendgrid.net/) failed: Name does not resolve

After appending `?region=global` to the existing DSN it did work again.

Reply:
Can you confirm that #59099 fixes this?

Reply:
Ah, missed that in my search. Yes that fixes it, thanks!